tests/flows/erc20_token_staking.go

```diff
@@ -0,0 +1,183 @@
+
+		// Construct a SubnetValidatorRegistrationMessage Warp message from the P-Chain
+		registrationSignedMessage := utils.ConstructSubnetValidatorRegistrationMessage(
+			validationID,
+			true,
+			subnetAInfo,
+			pChainInfo,
+			network,
+			signatureAggregator,
+		)
+
+		// Deliver the Warp message to the subnet
+		receipt = utils.CallERC20CompleteValidatorRegistration(
+			fundedKey,
+			subnetAInfo,
+			stakingManagerAddress,
+			registrationSignedMessage,
+		)
+		// Check that the validator is registered in the staking contract
+		registrationEvent, err := utils.GetEventFromLogs(
+			receipt.Logs,
+			stakingManager.ParseValidationPeriodRegistered,
+		)
+		Expect(err).Should(BeNil())
+		Expect(registrationEvent.ValidationID[:]).Should(Equal(validationID[:]))
+	}
+
+	//
+	// Delist the validator
+	//
+	{
+		receipt := utils.CallERC20InitializeEndValidation(
+			fundedKey,
+			subnetAInfo,
+			stakingManager,
+			validationID,
+		)
+		validatorRemovalEvent, err := utils.GetEventFromLogs(
+			receipt.Logs,
+			stakingManager.ParseValidatorRemovalInitialized,
+		)
+		Expect(err).Should(BeNil())
+		Expect(validatorRemovalEvent.ValidationID[:]).Should(Equal(validationID[:]))
+		Expect(validatorRemovalEvent.StakeAmount.Uint64()).Should(Equal(weight))
+
+		// Gather subnet-evm Warp signatures for the SetSubnetValidatorWeightMessage & relay to the P-Chain
+		// (Sending to the P-Chain will be skipped for now)
+		signedWarpMessage := network.ConstructSignedWarpMessage(context.Background(), receipt, subnetAInfo, pChainInfo)
+		Expect(err).Should(BeNil())
+		// Validate the Warp message, (this will be done on the P-Chain in the future)
+		msg, err := warpPayload.ParseAddressedCall(signedWarpMessage.UnsignedMessage.Payload)
+		Expect(err).Should(BeNil())
+		// Check that the addressed call payload is a registered Warp message type
+		var payloadInterface warpMessages.Payload
+		ver, err := warpMessages.Codec.Unmarshal(msg.Payload, &payloadInterface)
+		Expect(err).Should(BeNil())
+		registerValidatorPayload, ok := payloadInterface.(*warpMessages.SetSubnetValidatorWeight)
+		Expect(ok).Should(BeTrue())
+
+		Expect(ver).Should(Equal(uint16(warpMessages.CodecVersion)))
+		Expect(registerValidatorPayload.ValidationID).Should(Equal(validationID))
+		Expect(registerValidatorPayload.Weight).Should(Equal(uint64(0)))
+		Expect(registerValidatorPayload.Nonce).Should(Equal(uint64(0)))
+
+		// Construct a SubnetValidatorRegistrationMessage Warp message from the P-Chain
+		registrationSignedMessage := utils.ConstructSubnetValidatorRegistrationMessage(
+			validationID,
+			false,
+			subnetAInfo,
+			pChainInfo,
+			network,
+			signatureAggregator,
+		)
+
+		// Deliver the Warp message to the subnet
+		receipt = utils.CallERC20CompleteEndValidation(
+			fundedKey,
+			subnetAInfo,
+			stakingManagerAddress,
+			registrationSignedMessage,
+		)
+
+		// Check that the validator is registered in the staking contract
```

Comment:
The validator is still registered because they haven't collected their stake yet?

Reply:
This comment is incorrect. Fixed now.